### PR TITLE
KAFKA-14839: Exclude protected variable from JavaDocs

### PR DESCRIPTION
**Pull request by atu-sharm**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

Adding rules to display only Public API.

### CODE REVIEW

**Comment by atu-sharm** [comment]
@mjsax @machi1990 can you review

File: build.gradle
Comment (machi1990) [inline build.gradle]: @mjsax @atu-sharm  this is only applied to the streams project, and I am wondering if it should be a global config for all the projects e.g `clients` project: https://github.com/apache/kafka/blob/ca1bee3118dc790d122959faec2ca89716e08021/build.gradle#L1432 as well? What do you think?

Comment (atu-sharm) [reply build.gradle]: Yes, i agree @machi1990, @mjsax any thoughts?

Comment (mjsax) [reply build.gradle]: \cc @ijuma WDYT?

**Review by vvcephei — APPROVED** [review]
Thanks, @mjsax !

File: build.gradle
Comment (vvcephei) [reply build.gradle]: I'd be +1 on expanding the scope to the whole project. If something needs to be publicly documented, it should probably be made public.

FWIW, I'm also ok with merging as-is, and following up with a proposal to apply it to the whole repo.

**Comment by vvcephei** [comment]
Actually, @mjsax , I do have one question before merging: Are there classes that are meant to be overridden by users? If so, should their protected members be documented?

**Comment by mjsax** [comment]
Thank @vvcephei -- for K14839 in particular, it's nothing users should overwrite, in fact, they should not even extend the class in question at all.

I did not double check in general though, but I don't think we would have any `protected` vars for public interfaces/classed that uses would need to know of.

File: build.gradle
Comment (mjsax) [reply build.gradle]: @atu-sharm -- Leave it up to you, to change this PR or do a follow up. Just let me know what you prefer.

**Comment by atu-sharm** [comment]
Thanks, @vvcephei @mjsax @machi1990 for your valuable suggestions, i think it's better to apply this property to the whole project, made changes according to that

**Comment by mjsax** [comment]
Thanks for the PR! Merged to `trunk` and cherry-picked to `3.5` branch. \cc @mimaison

**Pull request merged**
